chore: updated gitconfig documentation to use the new command ensemble (#306)
epinio/docs: docs/references/git_configuration.md

# Git Configuration

## Overview

Starting with version **1.10.0**, Epinio supports Git configurations.

Configurations enable cloning of private repositories, disabling of SSL verification, and/or
extending verification through a custom bundle of certificates.

This is done on a per git host (+user/org, +repository) basis.

Management, including creation, is done through the
[epinio gitconfig](commands/cli/gitconfig/epinio_gitconfig.md)
command ensemble.

## Matching process

When importing from a git repository Epinio will use the most specific matching configuration, if
there is any.

This means that a matching configuration specifying url, user/organization, and repository has
priority over matching configurations specifying only url and user/organization, or even just the
url.

If no configuration is found then the cloning from the Git repository will run without any
customization.

## Github/Gitlab specialities

The public Github and Gitlab mega repositories support the use of a `PAT` (Personal Access Token)
over a plain combination of user and password.

When using a PAT it has to be set as the password, and the user can be set to anything except empty.

:::tip

> It is useful to set it to the username used to generate the token.

Author: Applied suggestion, will add a reason.

:::

## Detailed specification

:::note

> Regular users have no need to know the information in this section.
> It is useful to operators though, helping in debugging and inspection.

Reviewer: ```suggestion
This section contains information useful to operators for debugging and inspection. 
Regular users should not normally need to consult this section.
```
